Bio/SeqFeature.py

```diff
@@ -1004,22 +1006,26 @@ def __eq__(self, other):
         )
 
     def _shift(self, offset):
-        """Return a copy of the FeatureLocation shifted by an offset (PRIVATE)."""
+        """Return a copy of the FeatureLocation shifted by an offset (PRIVATE).
+
+        Returns self when location is relative to an external reference.
+        """
         # TODO - What if offset is a fuzzy position?
         if self.ref or self.ref_db:
-            # TODO - Return self?
-            raise ValueError("Feature references another sequence.")
+            return self
         return FeatureLocation(
             start=self._start._shift(offset),
             end=self._end._shift(offset),
             strand=self.strand,
         )
 
     def _flip(self, length):
-        """Return a copy of the location after the parent is reversed (PRIVATE)."""
+        """Return a copy of the location after the parent is reversed (PRIVATE).
+
+        Returns self when location is relative to an external reference.
+        """
         if self.ref or self.ref_db:
-            # TODO - Return self?
-            raise ValueError("Feature references another sequence.")
+            self
```

Comment:
return self

Reply:
Thank you. I'm slightly surprised flake8 didn't complain about that.